no-irregular-whitespace will give an error to irregular-whitespace if skipTemplates is true but template literal line breaks are included

<!--
    ESLint adheres to the Open JS Foundation Code of Conduct:
    https://eslint.org/conduct

    This template is for bug reports. If you are here for another reason, please see below:

    1. To propose a new rule: https://eslint.org/docs/developer-guide/contributing/new-rules
    2. To request a rule change: https://eslint.org/docs/developer-guide/contributing/rule-changes
    3. To request a change that is not a bug fix, rule change, or new rule: https://eslint.org/docs/developer-guide/contributing/changes
    4. If you have any questions, please stop by our chatroom: https://eslint.org/chat

    Note that leaving sections blank will make it difficult for us to troubleshoot and we may have to close the issue.
-->


**Tell us about your environment**

<!--
    If you are using ESLint v6.5.0 or later, you can run ESLint with the `--env-info` flag and paste the output here.
-->

* **ESLint Version: `7.13.0`**
* **Node Version: `v12.18.0`**
* **npm Version: `v6.14.4`**

**What parser are you using?: `default`**

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

<!-- Paste your configuration below: -->
```json
{
    "extends": ["eslint:recommended"],
    "env": {
        "es6": true,
        "browser": true,
        "jest": true,
        "node": true
    },
    "rules": {
        "no-unused-vars": "off",
        "no-irregular-whitespace": [
            "error",
            {
                "skipTemplates": true
            }
        ]
    }
}

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

<!-- Paste the source code below: -->
```js
const passed = `hoge　fuga`;
const error = `hoge　fuga
hoge　fuga`;

// \U3000 is included between all `hoge` and `fuga`.
// But ESLint only gives an error on the 3rd line.
```

<!-- Paste the command you used to run ESLint: -->
```bash
yarn eslint .\src\no-irregular-whitespace.js
```

**What did you expect to happen?**

ESLint does not raise an error for irregular-whitespace in template literals.

**What actually happened? Please include the actual, raw output from ESLint.**

```
  3:5  error  Irregular whitespace not allowed  no-irregular-whitespace
```

**Are you willing to submit a pull request to fix this bug?**

Probably difficult. I want to see if this is a bug or the correct behavior.

--- a/lib/rules/no-irregular-whitespace.js
+++ b/lib/rules/no-irregular-whitespace.js
@@ -82,7 +82,7 @@ module.exports = {
         const commentNodes = sourceCode.getAllComments();
 
         /**
-         * Removes errors that occur inside a string node
+         * Removes errors that occur inside the given node
          * @param {ASTNode} node to check for matching errors.
          * @returns {void}
          * @private
@@ -91,14 +91,12 @@ module.exports = {
             const locStart = node.loc.start;
             const locEnd = node.loc.end;
 
-            errors = errors.filter(({ loc: { start: errorLoc } }) => {
-                if (errorLoc.line >= locStart.line && errorLoc.line <= locEnd.line) {
-                    if (errorLoc.column >= locStart.column && (errorLoc.column <= locEnd.column || errorLoc.line < locEnd.line)) {
-                        return false;
-                    }
-                }
-                return true;
-            });
+            errors = errors.filter(({ loc: { start: errorLocStart } }) => (
+                errorLocStart.line < locStart.line ||
+                errorLocStart.line === locStart.line && errorLocStart.column < locStart.column ||
+                errorLocStart.line === locEnd.line && errorLocStart.column >= locEnd.column ||
+                errorLocStart.line > locEnd.line
+            ));
         }
 
         /**

--- a/tests/lib/rules/no-irregular-whitespace.js
+++ b/tests/lib/rules/no-irregular-whitespace.js
@@ -164,6 +164,12 @@ ruleTester.run("no-irregular-whitespace", rule, {
         { code: "`\u205f`", options: [{ skipTemplates: true }], parserOptions: { ecmaVersion: 6 } },
         { code: "`\u3000`", options: [{ skipTemplates: true }], parserOptions: { ecmaVersion: 6 } },
 
+        { code: "const error = ` \u3000 `;", options: [{ skipTemplates: true }], parserOptions: { ecmaVersion: 6 } },
+        { code: "const error = `\n\u3000`;", options: [{ skipTemplates: true }], parserOptions: { ecmaVersion: 6 } },
+        { code: "const error = `\u3000\n`;", options: [{ skipTemplates: true }], parserOptions: { ecmaVersion: 6 } },
+        { code: "const error = `\n\u3000\n`;", options: [{ skipTemplates: true }], parserOptions: { ecmaVersion: 6 } },
+        { code: "const error = `foo\u3000bar\nfoo\u3000bar`;", options: [{ skipTemplates: true }], parserOptions: { ecmaVersion: 6 } },
+
         // Unicode BOM.
         "\uFEFFconsole.log('hello BOM');"
     ],
@@ -538,6 +544,110 @@ ruleTester.run("no-irregular-whitespace", rule, {
                     type: "Program",
                     line: 1,
                     column: 14
+                }
+            ]
+        },
+        {
+            code: "\u3000\n`\u3000template`",
+            options: [{ skipTemplates: true }],
+            parserOptions: { ecmaVersion: 6 },
+            errors: [
+                {
+                    messageId: "noIrregularWhitespace",
+                    type: "Program",
+                    line: 1,
+                    column: 1
+                }
+            ]
+        },
+        {
+            code: "\u3000\n`\u3000multiline\ntemplate`",
+            options: [{ skipTemplates: true }],
+            parserOptions: { ecmaVersion: 6 },
+            errors: [
+                {
+                    messageId: "noIrregularWhitespace",
+                    type: "Program",
+                    line: 1,
+                    column: 1
+                }
+            ]
+        },
+        {
+            code: "\u3000`\u3000template`",
+            options: [{ skipTemplates: true }],
+            parserOptions: { ecmaVersion: 6 },
+            errors: [
+                {
+                    messageId: "noIrregularWhitespace",
+                    type: "Program",
+                    line: 1,
+                    column: 1
+                }
+            ]
+        },
+        {
+            code: "\u3000`\u3000multiline\ntemplate`",
+            options: [{ skipTemplates: true }],
+            parserOptions: { ecmaVersion: 6 },
+            errors: [
+                {
+                    messageId: "noIrregularWhitespace",
+                    type: "Program",
+                    line: 1,
+                    column: 1
+                }
+            ]
+        },
+        {
+            code: "`\u3000template`\u3000",
+            options: [{ skipTemplates: true }],
+            parserOptions: { ecmaVersion: 6 },
+            errors: [
+                {
+                    messageId: "noIrregularWhitespace",
+                    type: "Program",
+                    line: 1,
+                    column: 12
+                }
+            ]
+        },
+        {
+            code: "`\u3000multiline\ntemplate`\u3000",
+            options: [{ skipTemplates: true }],
+            parserOptions: { ecmaVersion: 6 },
+            errors: [
+                {
+                    messageId: "noIrregularWhitespace",
+                    type: "Program",
+                    line: 2,
+                    column: 10
+                }
+            ]
+        },
+        {
+            code: "`\u3000template`\n\u3000",
+            options: [{ skipTemplates: true }],
+            parserOptions: { ecmaVersion: 6 },
+            errors: [
+                {
+                    messageId: "noIrregularWhitespace",
+                    type: "Program",
+                    line: 2,
+                    column: 1
+                }
+            ]
+        },
+        {
+            code: "`\u3000multiline\ntemplate`\n\u3000",
+            options: [{ skipTemplates: true }],
+            parserOptions: { ecmaVersion: 6 },
+            errors: [
+                {
+                    messageId: "noIrregularWhitespace",
+                    type: "Program",
+                    line: 3,
+                    column: 1
                 }
             ]
         },

--- a/tests/lib/rules/no-irregular-whitespace.js
+++ b/tests/lib/rules/no-irregular-whitespace.js
@@ -164,6 +164,7 @@ ruleTester.run("no-irregular-whitespace", rule, {
         { code: "`\u205f`", options: [{ skipTemplates: true }], parserOptions: { ecmaVersion: 6 } },
         { code: "`\u3000`", options: [{ skipTemplates: true }], parserOptions: { ecmaVersion: 6 } },
 
+        { code: "`\u3000${foo}\u3000`", options: [{ skipTemplates: true }], parserOptions: { ecmaVersion: 6 } },
         { code: "const error = ` \u3000 `;", options: [{ skipTemplates: true }], parserOptions: { ecmaVersion: 6 } },
         { code: "const error = `\n\u3000`;", options: [{ skipTemplates: true }], parserOptions: { ecmaVersion: 6 } },
         { code: "const error = `\u3000\n`;", options: [{ skipTemplates: true }], parserOptions: { ecmaVersion: 6 } },
@@ -544,6 +545,19 @@ ruleTester.run("no-irregular-whitespace", rule, {
                     type: "Program",
                     line: 1,
                     column: 14
+                }
+            ]
+        },
+        {
+            code: "`something ${10\u3000} another thing`",
+            options: [{ skipTemplates: true }],
+            parserOptions: { ecmaVersion: 6 },
+            errors: [
+                {
+                    messageId: "noIrregularWhitespace",
+                    type: "Program",
+                    line: 1,
+                    column: 16
                 }
             ]
         },

